iot/api-client/manager/cloudiot_device_manager_example.py

```diff
@@ -0,0 +1,330 @@
+            parent=self.parent, body=registry_info, id=registry_id)
+
+        try:
+            response = request.execute()
+            print('Created registry', registry_id)
+            print(response)
+        except HttpError as e:
+            if e.resp.status == 409:
+                # Device registry already exists
+                print(
+                        'Registry', registry_id,
+                        'already exists - looking it up instead.')
+                request = self._service.projects().locations().registries(
+                        ).get(name=self.full_name)
+                request.execute()
+
+            else:
+                raise
+
+    def delete(self):
+        """Delete this registry."""
+        request = self._service.projects().locations().registries().delete(
+                name=self.full_name)
+        return request.execute()
+
+    def list_devices(self):
+        """List all devices in the registry."""
+        request = self._service.projects().locations().registries().devices(
+                ).list(parent=self.full_name)
+        response = request.execute()
+        return response.get('devices', [])
+
+    def _create_device(self, device_id, device_template):
+        request = self._service.projects().locations().registries().devices(
+        ).create(parent=self.full_name, body=device_template, id=device_id)
+        return request.execute()
+
+    def create_device_with_rs256(self, device_id, certificate_file):
+        """Create a new device with the given id, using RS256 for
+        authentication."""
+        with open(certificate_file) as f:
+            certificate = f.read()
+
+        # Create a device with the given certificate. Note that you can have
+        # multiple credentials associated with a device.
+        device_template = {
+            'credentials': [{
+                'publicKey': {
+                    'format': 'RSA_X509_PEM',
+                    'key': certificate
+                }
+            }]
+        }
+        return self._create_device(device_id, device_template)
+
+    def create_device_with_es256(self, device_id, public_key_file):
+        """Create a new device with the given id, using ES256 for
+        authentication."""
+        with open(public_key_file) as f:
+            public_key = f.read()
+
+        # Create a device with the given public key. Note that you can have
+        # multiple credentials associated with a device.
+        device_template = {
+            'credentials': [{
+                'publicKey': {
+                    'format': 'ES256_PEM',
+                    'key': public_key
+                }
+            }]
+        }
+        return self._create_device(device_id, device_template)
+
+    def create_device_with_no_auth(self, device_id):
+        """Create a new device with no authentication."""
+        return self._create_device(device_id, {})
+
+    def patch_es256_for_auth(self, device_id, public_key_file):
```

Comment:
Recommendation: name this after the *intent*, not the implementation, so maybe `add_es256_key_to_device`.